Remove git-url-parse requirement

philipsd6 opened the pull request:
Fixes #72

- replace `giturlparse.parse` with local implementation based on
  `urllib.parse.urlparse`
- Strip tilde from owner for Bitbucket Server user repos
- put `src_dir` and `lock dir` under directory named after the repo
  hostname
- remove `_get_lock_file()`

retr0h commented on gilt/config.py:
```diff
@@ -59,6 +63,35 @@ def config(filename):
     return [Config(**d) for d in _get_config_generator(filename)]
 
 
+def _parse_repo_uri(uri, scm="git"):
```
We are going to need unit tests for this.  At a minimum ensuring it has parity with the library it is replacing.

philipsd6 replied:
Done in latest update to this PR!

philipsd6 commented:
@retr0h I've added all the repo URIs from the git-url-parse tests, along with a couple more specific to Bitbucket Server. I added a new test to validate the repo URIs, and updated the existing tests to take into account the lock and clone files being in a hostname subdirectory. All tests pass.

jayvdb commented:
Not wanting to change your course, however there is discussion about new maintenance of git-url-parse at https://github.com/retr0h/git-url-parse/pull/39

@coala / @moremoban / @gitmateio depends on it in too many different projects for vendoring to be the ideal solution, and we do have a large-ish team, but we'd also be happy if someone else has a special interest in maintaining it.  fwiw, coala also needs the BitBucket support improved.

philipsd6 commented:
I don't much care one way or the other, but it seemed to me that if it was easy to remove the dependency on an unmaintained project then why not. But given that @retr0h isn't reviewing this PR after I made the requested changes, I've dropped my interest in this project entirely.

retr0h commented:
> But given that @retr0h isn't reviewing this PR after I made the requested changes, I've dropped my interest in this project entirely.

I haven't had a chance to look at this.  Sorry about that.

philipsd6 commented:
> > But given that @retr0h isn't reviewing this PR after I made the requested changes, I've dropped my interest in this project entirely.
> 
> I haven't had a chance to look at this. Sorry about that.

It's not personal -- we all have day jobs and sometimes other things take over our lives. I just think this project is getting very low attention, and we can't rely on it, so we'll come up with a different solution for our needs.

jayvdb commented:
@philipsd6 , depending on your needs, you may be interested in https://github.com/moremoban/moban , which isnt as mature as gilt at the things gilt does well, but it is maintained and has a few different features.  Another one I have seen but not tried is https://github.com/jpscaletti/copier .  Or you may someone more suitable in the lists at https://github.com/ksdme/org-status/issues/6 and https://gitlab.com/coala/package_manager/issues/237

retr0h commented:
> @philipsd6 , depending on your needs, you may be interested in https://github.com/moremoban/moban , which isnt as mature as gilt at the things gilt does well, but it is maintained and has a few different features. Another one I have seen but not tried is https://github.com/jpscaletti/copier . Or you may someone more suitable in the lists at [ksdme/org-status#6](https://github.com/ksdme/org-status/issues/6) and https://gitlab.com/coala/package_manager/issues/237

Gilt is maintained.  However, I am in the process of moving Gilt into my own repo.  I recently moved teams, and no longer have access to merge Gilt PRs.  I am hoping to straighten that out soon, and get the project back in line.

Again, my apologies.

ryandaugherty commented:
@philipsd6 your fix works great! I installed it in pip with the following command (somehow got errors, but it installed), and now gilt works with Bitbucket. Thanks much! 

`pip3 install git+https://github.com/philipsd6/gilt.git@fix/remove-giturlparse#egg=gilt`

@retr0h do you have access to merge yet? :)

retr0h commented:
> @retr0h do you have access to merge yet? Or move the repo? :)

Hoping I will :) 

/cc @cburgess

retr0h commented:
@ryandaugherty ok just got access back :)

ryandaugherty commented:
Yay! thanks!
